onnxruntime/core/dlpack/dlpack_converter.cc

```diff
@@ -223,14 +223,19 @@ OrtValue DlpackToOrtValue(DLManagedTensor* dlpack, bool is_bool_tensor) {
   ORT_ENFORCE(IsContiguousTensor(dlpack->dl_tensor), "ORT only supports contiguous tensor for now.");
   OrtDevice device = GetOrtDevice(dlpack->dl_tensor.ctx);
   MLDataType data_type = GetOrtValueDataType(dlpack->dl_tensor.dtype, is_bool_tensor);
-  std::function<void(void*)> deleter = [dlpack](void*) { dlpack->deleter((dlpack)); };
   OrtMemoryInfo info(GetOrtDeviceName(device), OrtDeviceAllocator, device, device.Id());
   std::unique_ptr<Tensor> p_tensor = std::make_unique<Tensor>(
       data_type, TensorShape(dlpack->dl_tensor.shape, static_cast<size_t>(dlpack->dl_tensor.ndim)),
       dlpack->dl_tensor.data, info);
 
   OrtValue ort_value;
-  ort_value.Init(p_tensor.release(), DataTypeImpl::GetType<Tensor>(), deleter);
+  auto tensor_type = DataTypeImpl::GetType<Tensor>();
+  std::function<void(void*)> deleter = [dlpack, tensor_type](void* p) {
+    dlpack->deleter((dlpack));
+    tensor_type->GetDeleteFunc()(p);
+  };
```

Comment:
Does this slightly simpler code work? It probably doesn't make a big difference, but it seems a pity to allocate space in the closure for a value that is always the same.

```suggestion
  std::function<void(void*)> deleter = [dlpack](void* p) {
    dlpack->deleter((dlpack));
    DataTypeImpl::GetType<Tensor>()->GetDeleteFunc()(p);
  };
```

(Even better would be to make it `static`, but I'm not sure if that's legal.)